components/helpers/post-via-iframe.js

```diff
@@ -1,63 +1,36 @@
     /**
-    * Converts a JSON schema+data into valid CSV string
-    * @param {object} schema - json schema used to render column headers
-    * @param {obejct} data - data array contining items to render as rows
-    */
-    function jsonToCSV(schema, data = [], separator = ' ,', lineTerminator = '\n') {
-        // eslint-disable-next-line no-param-reassign
-        schema = ((schema && schema.properties) ? schema.properties : schema || {});
-
-        // TECH-DEBT - sort out this throwing..
-        // eslint-disable-next-line no-throw-literal
-        if (separator.indexOf('"') > -1) throw 'Invalid separator';
-        // eslint-disable-next-line no-throw-literal
-        if (lineTerminator.indexOf('"') > -1) throw 'Invalid lineTerminator';
-
-        const keys = [];
-        let line = [];
-        const rows = [];
-        const headers = [];
-
-        // wraps any value containing separators in quotes
-        // need to also escape double quotes
-        const escapeValue = function (value) {
-            if (value.indexOf(separator) > -1 || value.indexOf(lineTerminator) > -1) {
-                return `"${value.replace(/"/g, '"')}"`;
-            }
-            return value;
-        };
-
-        // get keys we're interested in
-        Object.keys(schema).forEach(key => {
-            if (Object.prototype.hasOwnProperty.call(schema, key) && key !== 'links') {
-                keys.push(key);
-            }
+      * Sends data to the server via a form post in a hidden frame
+      * @param {string} url - url to post the data to
+      * @param {object} data - JSON key/value object - a hidden field is created for each key
+      */
+    function postViaIframe(url, data) {
+        const targetName = `_post_via_iframe_${Date.now()}`;
+        // eslint-disable-next-line no-used-vars
+        const iframe = createElement(document.body, 'iframe', {
+            src: '',
+            name: targetName,
+            style: 'display: none; visibility: hidden;'
         });
-
-        // build headers
-        keys.forEach(key => {
-            headers.push(schema[key].title || key);
+        const form = createElement(document.body, 'form', {
+            action: url,
+            enctype: 'application/x-www-form-urlencoded',
+            method: 'POST',
+            target: targetName,
+            style: 'display: none !important'
         });
 
-        // add headers as first rows item
-        rows.push(headers.join(separator));
-
-        // build rows
-        data.forEach(item => {
-
-            line = []; // create new line collection
-
-            // add each data item to the current line
-            keys.forEach(key => {
-                line.push(escapeValue(item[key] || ''));
-            });
-
-            // convert line into CSV row and add to rows collection
-            rows.push(line.join(separator));
-        });
+        // eslint-disable-next-line no-restricted-syntax
+        for (const key in data) {
```

Comment:
Could use Object.keys(data).forEach() to avoid the eslint-disable, but we can leave that for refactor now.